src/Compilers/Test/Core/Traits/WorkItemTraitDiscoverer.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities;
+
+public sealed class WorkItemTraitDiscoverer : ITraitDiscoverer
+{
+    public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+    {
+        var arguments = traitAttribute.GetConstructorArguments().ToArray();
+        if (arguments is [int id, string url])
+        {
+            yield return new KeyValuePair<string, string>("WorkItemId", id.ToString());
+            yield return new KeyValuePair<string, string>("WorkItemUrl", url);
+        }
+        else if (arguments is [string onlyUrl])
+        {
+            yield return new KeyValuePair<string, string>("WorkItemUrl", onlyUrl);
```

Comment:
In the case where the `WorkItem` only contains a URL (which is what we now commonly do), maybe we should extract the ID and make it available to search on too?

Reply:
Even if there is just `WorkItem(url)` attribute, searching for `Trait:IssueId` works as well (i.e., a substring match is enough)